Source/Core/VideoBackends/Vulkan/VKSwapChain.cpp

```diff
@@ -36,8 +36,174 @@ SwapChain::~SwapChain()
   DestroySurface();
 }
 
-VkSurfaceKHR SwapChain::CreateVulkanSurface(VkInstance instance, const WindowSystemInfo& wsi)
+VkSurfaceKHR SwapChain::CreateVulkanSurface(VkInstance instance, VkPhysicalDevice physical_device,
+                                            const WindowSystemInfo& wsi)
 {
+#if defined(VK_USE_PLATFORM_DISPLAY_KHR)
+  if (wsi.type == WindowSystemType::DRM)
+  {
+    // Get the first display
+    uint32_t display_count = 1;
+    VkDisplayPropertiesKHR display_props;
+    if (VkResult err = vkGetPhysicalDeviceDisplayPropertiesKHR(physical_device, &display_count,
+                                                               &display_props);
+        err != VK_SUCCESS && err != VK_INCOMPLETE)
+    {
+      LOG_VULKAN_ERROR(err, "vkGetPhysicalDeviceDisplayPropertiesKHR failed: ");
+      return VK_NULL_HANDLE;
+    }
+
+    // Get the first mode of the display
+    uint32_t mode_count = 0;
+    if (VkResult err = vkGetDisplayModePropertiesKHR(physical_device, display_props.display,
+                                                     &mode_count, nullptr);
+        err != VK_SUCCESS)
+    {
+      LOG_VULKAN_ERROR(err, "vkGetDisplayModePropertiesKHR failed: ");
+      return VK_NULL_HANDLE;
+    }
+
+    if (mode_count == 0)
+    {
+      ERROR_LOG_FMT(VIDEO, "Cannot find any mode for the display!");
+      return VK_NULL_HANDLE;
+    }
+
+    VkDisplayModePropertiesKHR mode_props;
+    mode_count = 1;
+    if (VkResult err = vkGetDisplayModePropertiesKHR(physical_device, display_props.display,
+                                                     &mode_count, &mode_props);
+        err != VK_SUCCESS && err != VK_INCOMPLETE)
+    {
+      LOG_VULKAN_ERROR(err, "vkGetDisplayModePropertiesKHR failed: ");
+      return VK_NULL_HANDLE;
+    }
+
+    // Get the list of planes
+    uint32_t plane_count = 0;
+    if (VkResult err =
+            vkGetPhysicalDeviceDisplayPlanePropertiesKHR(physical_device, &plane_count, nullptr);
+        err != VK_SUCCESS)
+    {
+      LOG_VULKAN_ERROR(err, "vkGetPhysicalDeviceDisplayPlanePropertiesKHR failed: ");
+      return VK_NULL_HANDLE;
+    }
+
+    if (plane_count == 0)
+    {
+      ERROR_LOG_FMT(VIDEO, "No display planes found!");
+      return VK_NULL_HANDLE;
+    }
+
+    // Find a plane compatible with the display
+    // Find a plane compatible with the display
+    uint32_t compatible_plane_index = UINT32_MAX;
```

Comment:
I think once is enough 😅 
```suggestion
    // Find a plane compatible with the display
    uint32_t compatible_plane_index = UINT32_MAX;
```

Reply:
Good catch😸